src/BuildPrediction/Predictors/ProjectFileAndImportsGraphPredictor.cs

```diff
@@ -0,0 +1,41 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+namespace Microsoft.Build.Prediction.Predictors
+{
+    using System.Collections.Generic;
+    using Microsoft.Build.Graph;
+
+    /// <summary>
+    /// Finds project filename and imports from transitive dependencies as inputs.
+    /// </summary>
+    public sealed class ProjectFileAndImportsGraphPredictor : IProjectGraphPredictor
+    {
+        /// <inheritdoc/>
+        public void PredictInputsAndOutputs(ProjectGraphNode projectGraphNode, ProjectPredictionReporter predictionReporter)
```

Comment:
Interesting version issue here. I wonder if it's worth documenting what versions of MSBuild this works with? I suppose you could use the predictor library without this rule on 16.4 and below, but this one requires 16.5, right? Or maybe worry about that later.

Reply:
The package has a declared dependency on a specific MSBuild package, so that should be sufficient for determining the min version of MSBuild. Supporting multiple versions would be pretty hard with how nuget dependencies work.